lib/FusionInventory/Agent/Task/Inventory/Generic/Dmidecode/Battery.pm

```diff
@@ -18,33 +18,40 @@ sub doInventory {
     my $inventory = $params{inventory};
     my $logger    = $params{logger};
 
-    my $battery = _getBattery(logger => $logger);
+    my @batteries = _getBatteries(logger => $logger);
 
-    return unless $battery;
+    return unless @batteries;
 
-    $inventory->addEntry(
-        section => 'BATTERIES',
-        entry   => $battery
-    );
+    _mergeBatteries($inventory, @batteries);
 }
 
-sub _getBattery {
+sub _getBatteries {
     my $infos = getDmidecodeInfos(@_);
 
     return unless $infos->{22};
 
-    my $info    = $infos->{22}->[0];
+    my $batteries;
+    for my $info (@{$infos->{22}}) {
+        my $data = _extractBatteryData($info);
+        push @$batteries, $data if $data;
+    }
+
+    return (scalar @$batteries) > 0 ? $batteries : undef ;
```

Comment:
more readable construct (at least for me):
return $batteries ? @$batteries : undef;